chain-core/src/state/account.rs

```diff
@@ -22,8 +22,36 @@ pub type Count = u64;
 /// StakedState update counter
 pub type Nonce = u64;
 
-/// StakedState address type (TODO: enum?)
-pub type StakedStateAddress = RedeemAddress;
+/// StakedState address type
+#[derive(
+    Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord, Encode, Decode, Serialize, Deserialize,
+)]
+pub enum StakedStateAddress {
+    BasicRedeem(RedeemAddress),
+}
+
+impl TryFrom<&Vec<u8>> for StakedStateAddress {
```

Comment:
I think `TryFrom<T> for StakedStateAddress where T: AsRef<[u8]>` will be more flexible.

Reply:
I'll try if that works

Reply:
changed it to &[u8]